### ISSUE TITLE
Not scraping RD cloud

### ISSUE THREAD

**Issue by sathish804** [SwagOtaku/plugin.video.kaito#24]
Hi,
Good day!! 

I have added Naruto magnet to my RD cloud and in Kaito for RD enabled the option to scrape the rd cloud. But it always scrapping from Gogo and skipping RD cloud. 

Please look into this. 

Also please add more providers for scrapping.

Thanks in advance.

**Comment by DaisyF8** [comment]
Has the magnet finished downloading yet ?

**Comment by sathish804** [comment]
Yes its already finished downloading. Please see the attached screen shot. 
![Screenshot_20201227-152522](https://user-images.githubusercontent.com/14090242/103172968-dc481c00-4857-11eb-80a5-1b1b783be698.jpg)

**Comment by DaisyF8** [comment]
Naruto has 10 cached sources allready available. You should find these also.

I don't think your RD has been authenticated correctly. In addon-settings, between "Enable Real-Debrid" and "Authorize RD" there is a field "Real-Debrid username". 

Do you see your username there behind it? It appears if auth was successful.

**Comment by sathish804** [comment]
Yes auth is successful and I see my username in Kaito but it not scraping the RD cloud at all. 
![IMG_20210101_152157](https://user-images.githubusercontent.com/14090242/103440498-7f2cdb80-4c46-11eb-8d9c-615e35a55c92.jpg)

**Comment by SwagOtaku** [comment]
I have an update with improving cloud inspection ready to release that should fix this issue, but as i'm writing this trakt is having issues. So i'm holding out releasing until trakt stabilizes.

**Comment by SwagOtaku** [comment]
@sathish804 also set real-debrid priority to 10 instead of 1, higher number mean more priority

**Comment by sathish804** [comment]
@SwagOtaku I have only RD as debrid service, I tried with priority 10 and 1 in both the cases behavior is same.

**Comment by SwagOtaku** [comment]
@sathish804 released new update with improved cloud inspection

**Comment by sathish804** [comment]
@SwagOtaku Thank you, I will check it and let you know.

**Comment by sathish804** [comment]
@SwagOtaku I checked it for RD cloud cached sources but still the same issue.
This is the screen shot from Nyaa where we have full Naruto series. 
![Screenshot_20210103-100924](https://user-images.githubusercontent.com/14090242/103475650-7c012f00-4daf-11eb-97b4-8e399792bd4a.jpg)

This is the screen shot of same magnet cached in my rd cloud and downloaded. 
![Screenshot_20210103-100905](https://user-images.githubusercontent.com/14090242/103475662-963b0d00-4daf-11eb-881f-c4a4f69118e9.jpg)

This is the screen shot of Kaito source select which is not showing any RD sources. As shared in earlier screen shots of other post RD is already authorized in Kaito. 

![IMG_20210103_100844](https://user-images.githubusercontent.com/14090242/103475686-c97d9c00-4daf-11eb-9ec6-ffbe9afcda12.jpg)

Please let me know if you need any further information. 

Thanks and Regards, 
Sathish.

**Comment by sathish804** [comment]
@SwagOtaku One more update for this issue. If I have only one torrent cached for that title then Kaito is not finding it but if I have more than 1 then its finding it.

**Comment by TaiDak** [comment]
Ok, so after fiddling around with Premiumizer, I found why kaito can see some cached torrents, or files in your cloud storage. First off, this only applies to files in your cloud storage, as I yet don't know what kaito looks for in regular cached torrents.

In your cloud storage, in order for kaito to find an episode in a batch, the episode must be in the root of the batch folder, it cannot be inside another folder. So for example if you were to add a batch of say "Naruto" to your RD cloud, you have to make sure the episodes are not organized into separate folders by seasons, all the episodes have to be in the same folder, the root folder. THIS ONLY APPLIES TO CLOUD FILES, for some reason kaito will find episodes in season folders when kaito manages to find a cached torrent. Second thing, kaito will not see the episode file unless the has the the following symbols/word/letters 
" Episode, e, _, -, (" and 01 as episode number. Episode number must be at least 2 digits, not 1.
Examples of episode naming

Naruto Episode01 
Naruto E01
Naruto _01
Naruto -01
Naruto (01

I think this applies to both cached torrents, and torrents you yourself added to your RD cloud, or Premiumizer.

If the episode is part of a batch, it doesn't even need the series name as long as it has the symbols/word/letter in front of 01.

Example would be a folder name "Naruto" with the first episode inside being -01.mvk. 


You only need the title of the anime in the episode file if the episode file is a single file and not part of a batch.
For example for newer animes, since they're still being released, don't have a batch. So in order for kaito to find them, they need title, followed by the symbols, and then episode number. Example

Re zero episode 01.mvk
Or
Re zero - 01.mvk

Third thing I found out, is that if you have an incompatible torrent in your cloud with the same name as a compatible torrent, kaito will not find the compatible one. This means that if you have for example

Naruto>Season 1> Episode01.mvk and 
Naruto>Episode 01.mvk

Kaito will not see either one, you have to remove the incompatible torrent from your cloud, that is the one with the wrong folder structure "Naruto>Season 1> Episode 01.mvk has to be removed.
